azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java

```diff
@@ -0,0 +1,263 @@
+             * @return the next stage of the definition
+             */
+            DefineAgentPoolProfiles withSshKey(String sshKeyData);
+        }
+
+        /**
+         * The stage of the container service definition allowing to specific diagnostic settings.
+         */
+        interface WithDiagnosticsProfile {
+            /**
+             * Enable diagnostics.
+             * @return the create stage of the definition
+             */
+            WithCreate withDiagnostics();
+
+            /**
+             * Disable diagnostics.
+             * @return the create stage of the definition
+             */
+            WithCreate withoutDiagnostics();
+        }
+
+        /**
+         * The stage of the definition which contains all the minimum required inputs for
+         * the resource to be created (via {@link WithCreate#create()}), but also allows
+         * for any other optional settings to be specified.
+         */
+        interface WithCreate extends
+                Creatable<ContainerService>,
+                Resource.DefinitionWithTags<WithCreate>,
+                ContainerService.DefinitionStages.WithOrchestratorProfile,
+                ContainerService.DefinitionStages.WithDiagnosticsProfile {
+        }
+    }
+
+    /**
+     * The template for an update operation, containing all the settings that
+     * can be modified.
+     * <p>
+     * Call {@link Update#apply()} to apply the changes to the resource in Azure.
+     */
+    interface Update extends
+            Resource.UpdateWithTags<Update>,
+            Appliable<ContainerService>,
+            ContainerService.UpdateStages.WithUpdateAgentPoolCount,
+            ContainerService.UpdateStages.WithDiagnosticsProfile {
+    }
+
+    /**
+     * Grouping of container service update stages.
+     */
+    interface UpdateStages {
+        /**
+         * The stage of the container service definition allowing to specific diagnostic settings.
+         */
+        interface WithDiagnosticsProfile {
+            /**
+             * Enable diagnostics.
+             * @return the next stage of the update
+             */
+            Update withDiagnostics();
+
+            /**
+             * Disable diagnostics.
+             * @return the next stage of the update
+             */
+            Update withoutDiagnostics();
+        }
+
+        /**
+         * The stage of the container service definition allowing to specific diagnostic settings.
+         */
+        interface WithUpdateAgentPoolCount {
+            /**
+             * Enable diagnostics.
+             * @param agentPoolCount the number of agents (VMs) to host docker containers.
+             *                       Allowed values must be in the range of 1 to 100 (inclusive).
+             *                       The default value is 1.
+             * @return the next stage of the update
+             */
+            Update withAgentPoolCount(int agentPoolCount);

```

Comment:
something feels wrong about this nanming. If I understand correctly, the pool is the collection of agents. This specifies the number of agents in a pool, not the number of pools, right?  In which case, shoulnd't this say `withAgentCount()`?